### PR TITLE
Registration tests

### PR DESCRIPTION
**Pull request by hturnbull93**

### CODE REVIEW

File: app/controllers/api/v1/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Api::V1::RegistrationsController < ApplicationController
+   
+  def create
+    begin
+      user = User.create!(
+        email: params["email"],
+        first_name: params["first_name"],
+        last_name: params["last_name"],
+        password: params["password"]
+      )
```
Comment (tommyrharper) [inline app/controllers/api/v1/registrations_controller.rb:10]: Nice params

File: app/controllers/api/v1/registrations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      if user 
+        session[:user_id] = user.id
+        render json: {
+          status: :created,
+          user: user
+        }
+      end
+    rescue => exception
```
Comment (tommyrharper) [inline app/controllers/api/v1/registrations_controller.rb:18]: cool use of rescue with the begin statement

File: app/controllers/api/v1/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            logged_in: true
+          }
+       else
+        render json: {
+            status: 401
+          }
+       end 
+    end
+
+    def destroy
```
Comment (tommyrharper) [inline app/controllers/api/v1/sessions_controller.rb]: So this logs people out

File: spec/controllers/registrations_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      last_name: "Gangi",
+      email: "samgangi@lotr.com",
+      password: "theshire"
+    }
+    json = JSON.parse(response.body)
+
+    expect(json['status']).to eq "created"
+    expect(json['user']['email']).to eq "samgangi@lotr.com"
+    expect(json['user']['first_name']).to eq "Sam"
+    expect(json['user']['last_name']).to eq "Gangi"
```
Comment (tommyrharper) [inline spec/controllers/registrations_controller_spec.rb:17]: Awesome tests!

File: spec/controllers/sessions_controller_spec.rb [hunk 5]
```diff
@@ -1,5 +1,48 @@
...
+  end
+
+  it "POST /sessions incorrect password, responds with json status 401 " do
+    post :create, params: { 
+      email: "samgangi@lotr.com",
+      password: "INCORRECT"
+    }
+    json = JSON.parse(response.body)
+
+    expect(json['status']).to eq 401
```
Comment (tommyrharper) [inline spec/controllers/sessions_controller_spec.rb]: more wonderful testing

**Comment by FayeCarter** [comment]
Very nice

**Review by FayeCarter — APPROVED** [review]

**Pull request merged**